client/app/components/ParameterValueInput.jsx

```diff
@@ -0,0 +1,149 @@
+import { isNull, isUndefined } from 'lodash';
+import React from 'react';
+import PropTypes from 'prop-types';
+import { react2angular } from 'react2angular';
+import { DateInput } from './DateInput.jsx';
+import { DateRangeInput } from './DateRangeInput.jsx';
+import { DateTimeInput } from './DateTimeInput.jsx';
+import { DateTimeRangeInput } from './DateTimeRangeInput.jsx';
+import { QueryBasedParameterInput } from './QueryBasedParameterInput.jsx';
+
+export function ParameterValueInput({
+  type,
+  value,
+  onSelect,
+  enumOptions,
+  queryId,
+  clientConfig, // eslint-disable-line react/prop-types
+  Query, // eslint-disable-line react/prop-types
+}) {
+  enumOptions = enumOptions.split('\n');
+
+  return (
+    <span>
+      {
+        (type === 'datetime-with-seconds') &&
+        <DateTimeInput
+          value={value}
+          onSelect={onSelect}
+          withSeconds
+          clientConfig={clientConfig}
+        />
+      }
+      {
+        (type === 'datetime-local') &&
+        <DateTimeInput
+          value={value}
+          onSelect={onSelect}
+          clientConfig={clientConfig}
+        />
+      }
+      {
+        (type === 'date') &&
+        <DateInput
+          value={value}
+          onSelect={onSelect}
+          clientConfig={clientConfig}
+        />
+      }
+      {
+        (type === 'datetime-range-with-seconds') &&
+        <DateTimeRangeInput
+          value={value}
+          onSelect={onSelect}
+          withSeconds
+          clientConfig={clientConfig}
+        />
+      }
+      {
+        (type === 'datetime-range') &&
+        <DateTimeRangeInput
+          value={value}
+          onSelect={onSelect}
+          clientConfig={clientConfig}
+        />
+      }
+      {
+        (type === 'date-range') &&
+        <DateRangeInput
+          value={value}
+          onSelect={onSelect}
+          clientConfig={clientConfig}
+        />
+      }
+      {
+        (type === 'enum') &&
+        <select
+          className="form-control"
+          value={isNull(value) || isUndefined(value) ? '' : value}
+          onChange={event => onSelect(event.target.value)}
+        >
+          {enumOptions.map(option => (
+            <option value={option} key={option}>{ option }</option>
+          ))}
+        </select>
+      }
+      {
+        (type === 'query') &&
+        <QueryBasedParameterInput
+          value={value}
+          queryId={queryId}
+          onSelect={onSelect}
+          Query={Query}
+        />
+      }
+      {
+        ((type === 'text') || (type === 'number')) &&
+        <input
+          type={type}
+          className="form-control"
+          value={isNull(value) || isUndefined(value) ? '' : value}
+          onChange={event => onSelect(event.target.value)}
+        />
+      }
+    </span>
```

Comment:
I think it will be cleaner and more maintainable to have multiple render functions. I.e. something like:

```javascript
class ParameterValueInput extends Component {
  renderQueryBasedInput() {}
  renderDateInput() {}
  ... more render functions ...
  render() {
    const renderers = { query: this.renderQueryBasedInput, date: this.renderDateInput, ...};
    return renderers[type]();
  }
}
```